indico/modules/rb_new/util.py

```diff
@@ -28,19 +28,26 @@
 from indico.core.db import db
 from indico.core.db.sqlalchemy.util.queries import db_dates_overlap, escape_like
 from indico.modules.rb import rb_is_admin, rb_settings
+from indico.modules.rb.models.blocked_rooms import BlockedRoom, BlockedRoomState
+from indico.modules.rb.models.blockings import Blocking
 from indico.modules.rb.models.equipment import EquipmentType, RoomEquipmentAssociation
 from indico.modules.rb.models.favorites import favorite_room_table
 from indico.modules.rb.models.reservation_occurrences import ReservationOccurrence
 from indico.modules.rb.models.reservations import RepeatFrequency, Reservation
 from indico.modules.rb.models.room_attributes import RoomAttributeAssociation
+from indico.modules.rb.models.room_bookable_hours import BookableHours
+from indico.modules.rb.models.room_nonbookable_periods import NonBookablePeriod
 from indico.modules.rb.models.rooms import Room
-from indico.modules.rb_new.schemas import reservation_occurrences_schema, rooms_schema
+from indico.modules.rb_new.schemas import (blockings_schema, bookable_hours_schema, nonbookable_periods_schema,
+                                           reservation_occurrences_schema, rooms_schema)
 from indico.util.caching import memoize_redis
+from indico.util.date_time import get_overlap
 from indico.util.struct.iterables import group_list
 
 
 BOOKING_TIME_DIFF = 20  # (minutes)
 DURATION_FACTOR = 0.25
+RESERVATION_OCCURRENCE_TMP = namedtuple('ReservationOccurrenceTmp', ('start_dt', 'end_dt', 'reservation'))
```

Comment:
`TempReservationOccurrence`? it's pretty much like a class so i'd use CamelCase for it